diff --git a/mlir/lib/Dialect/Transform/IR/CMakeLists.txt b/mlir/lib/Dialect/Transform/IR/CMakeLists.txt
index 5b4989f328e6..8652a352d694 100644
--- a/mlir/lib/Dialect/Transform/IR/CMakeLists.txt
+++ b/mlir/lib/Dialect/Transform/IR/CMakeLists.txt
@@ -17,6 +17,24 @@ add_mlir_dialect_library(MLIRTransformDialect
   MLIRRewrite
   MLIRSideEffectInterfaces
   MLIRTransforms
+  # Quite Odd that we need this here
+  MLIRArmSMETestPasses
+  MLIRArithTransforms
+  MLIRAsyncToLLVM
+  MLIRLinalgTransforms
+  MLIRMemRefTransforms
+  MLIRAffineTransforms
+  MLIRArithToLLVM
+  MLIROpenMPToLLVM
+  MLIRIndexToLLVM
+  MLIRComplexToLLVM
+  MLIRMathToLLVM
+  MLIRMathTransforms
+  MLIRAffineToStandard
+  MLIRReconcileUnrealizedCasts
+  MLIRAsyncTransforms
+  MLIRVectorToLLVM
+  MLIRVectorToLLVMPass
   MLIRTransformDialectInterfaces
   MLIRTransformDialectUtils
   )
